tokio/src/runtime/task/core.rs

```diff
@@ -80,6 +80,9 @@ unsafe impl Sync for Header {}
 pub(super) struct Trailer {
     /// Consumer task waiting on completion of this task.
     pub(super) waker: UnsafeCell<Option<Waker>>,
+    /// The tracing ID for this instrumented task.
+    #[cfg(all(tokio_unstable, feature = "tracing"))]
+    pub(super) id: Option<tracing::Id>,
```

Comment:
:+1: --- but maybe this should be called `span_id` or something so it's clear that it's specific to tracing? not important at all...

Reply:
one thought: since this is potentially accessed in all waker ops, I wonder if it's hot enough that it should go in the header rather than the trailer? IIRC, the motivation for separating task data into headers and trailers is that the header should be prefetched into cache when the task pointer is dereferenced, so we don't have to do additional memory accesses. 

currently, the waker ops only touch the header, so they are only touching data that should be prefetched. but now, they access both the header _and_ the trailer, which might break this property. and, since we check if the task has an ID before we emit the event, we'll access the trailer whenever Tokio is built with tracing enabled `--cfg tokio_unstable`, even if tracing is disabled at runtime. 

it would be great to hear from @carllerche on this?

Reply:
(re: header vs trailer) Yea, could be! I threw it in here since it was related to waker stuff, and the waker is in the trailer. But I also have not internalized all the `RawTask` stuff.

Reply:
Moved to header.